### PR TITLE
[libwebp] Do not emit AVX when not requested by the toolchain

### LINKED ISSUE #14568: [libwebp] Do not enable SSE4 support with /arch:AVX
The library source, in file [cmake/cpu.cmake:36](https://github.com/webmproject/libwebp/blob/master/cmake/cpu.cmake) sets **/arch:AVX** compiler flag to enable SSE4 support. It even has a comment stating it is intentional.
This 'ofcause enables the AVX instruction-set, resulting in a executable that will not run on a CPU without AVX.
e.q. function WebPDecode will crash with exception 0xC000001d 'Illegal Instruction' on a Intel Atom x5 CPU (build details in my case: VS2019, Release, x86, Cross-Compiled, both Win10).

The library should be clear about which instruction-set it is using, SSE4 or AVX.

As a workaround I propose an addition to the vcpkg-patch file: '0007-fix-arm-build.patch' that removes the '/arch:AVX' option.

```
diff --git a/cmake/cpu.cmake b/cmake/cpu.cmake
index da9a42a4..cb7a050d 100644
--- a/cmake/cpu.cmake
+++ b/cmake/cpu.cmake
@@ -33,7 +33,7 @@ set(WEBP_SIMD_FILE_EXTENSIONS
     "_sse41.c;_sse2.c;_mips32.c;_mips_dsp_r2.c;_neon.c;_msa.c")
 if(MSVC)
   # MSVC does not have a SSE4 flag but AVX support implies SSE4 support.
-  set(SIMD_ENABLE_FLAGS "/arch:AVX;/arch:SSE2;;;;")
+  set(SIMD_ENABLE_FLAGS "/arch:SSE2;;;;;")
   set(SIMD_DISABLE_FLAGS)
 else()
   set(SIMD_ENABLE_FLAGS
@@ -73,7 +73,7 @@ foreach(I_SIMD RANGE ${WEBP_SIMD_FLAGS_RANGE})
     set(CMAKE_REQUIRED_FLAGS ${SIMD_COMPILE_FLAG})
     webp_check_compiler_flag(${WEBP_SIMD_FLAG} ${WEBP_ENABLE_SIMD})
   else()
-    if(MSVC)
+    if(MSVC AND NOT (CMAKE_GENERATOR_PLATFORM MATCHES "^ARM"))
       list(GET SIMD_ENABLE_FLAGS ${I_SIMD} SIMD_COMPILE_FLAG)
     else()
       set(SIMD_COMPILE_FLAG " ")
```

### PR DESCRIPTION
**Pull request by JonLiu1993**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes #14568
The library source, in file cmake/cpu.cmake:36 sets /arch:AVX compiler flag to enable SSE4 support.This 'ofcause' enables the AVX instruction-set, resulting in a executable that will not run on a CPU without AVX.
As a workaround propose removes the '/arch:AVX' option.

### CODE REVIEW

**Comment by JackBoosY** [comment]
Closing this PR since it seems that no progress is being made. Please reopen if work is still being done.

**Comment by BillyONeal** [comment]
Thanks!

File: ports/libwebp/0007-fix-arm-build.patch [hunk 1]
```diff
@@ -1,13 +1,22 @@
...
-index bc0dbc9..381f64a 100644
+index bc0dbc9..f9b7327 100644
 --- a/cmake/cpu.cmake
 +++ b/cmake/cpu.cmake
+@@ -48,7 +48,7 @@ if(MSVC AND CMAKE_C_COMPILER_ID STREQUAL "MSVC")
+   if(MSVC_VERSION GREATER_EQUAL 1800 AND NOT CMAKE_C_FLAGS MATCHES "/arch:")
+     set(SIMD_ENABLE_FLAGS)
+   else()
+-    set(SIMD_ENABLE_FLAGS "/arch:AVX;/arch:SSE2;;;;")
++    set(SIMD_ENABLE_FLAGS "/arch:SSE2;;;;;")
```
Comment (dg0yt) [inline ports/libwebp/0007-fix-arm-build.patch:10]: This changes the position of the sse2 flag, but AFAIU this list must match other variables a few lines before:
~~~cmake
# those are included in the names of WEBP_USE_* in c++ code.
set(WEBP_SIMD_FLAGS "SSE41;SSE2;MIPS32;MIPS_DSP_R2;NEON;MSA")
set(WEBP_SIMD_FILE_EXTENSIONS
    "_sse41.c;_sse2.c;_mips32.c;_mips_dsp_r2.c;_neon.c;_msa.c")
~~~
Does this now use SSE41 instructions with SSE2 confiurations?

**Review by dg0yt — COMMENTED** [review]
This PR looked suspicious when I noticed it a few weeks ago. And now I looked at it again in the context of #32940.
So this patch was developed against 2021 source code for a 2020 bug, and it was merged now in 2023. Did anybody verify the changes which were made upstream in the meantime?
The patch still applying syntactically doesn't mean that it is still a valid modification (if it ever was valid).

File: ports/libwebp/0007-fix-arm-build.patch (hunk 1 above)
Comment (LilyWangLL) [reply ports/libwebp/0007-fix-arm-build.patch:10]: Ping @JonLiu1993

**Pull request merged**